user-management/src/main/scala/com/normation/plugins/usermanagement/UserManagementService.scala

```diff
@@ -102,52 +133,48 @@ object UserManagementService {
 
   def computeRoleCoverage(roles: Set[Role], authzs: Set[AuthorizationType]): Option[Set[Role]] = {
 
-    def compareRights(r: Role): Option[Role] = {
-      if (r.name == "no_rights") {
+    def compareRights(r: Role, as: Set[AuthorizationType]): Option[Role] = {
+      if (r == Role.NoRights) {
         None
       } else {
-        val authzNames     = authzs.map(_.id)
-        val roleAuthzNames = r.rights.authorizationTypes.map(_.id)
-        val commonRights   = roleAuthzNames.intersect(authzNames)
+        val commonRights = r.rights.authorizationTypes.intersect(as)
         commonRights match {
           // Intersection is total
-          case cr if cr == roleAuthzNames => Some(r)
-          case cr if cr.nonEmpty          =>
-            val test = cr.flatMap(RoleToRights.parseAuthz)
-            if (test.isEmpty) {
-              None
-            } else {
-              val c = Custom(new Rights(test.toSeq: _*))
-              Some(c)
-            }
-          case _                          => None
+          case cr if cr == r.rights.authorizationTypes => Some(r)
+          case cr if cr.nonEmpty                       =>
+            Some(Custom(new Rights(cr.toSeq: _*)))
+          case _                                       => None
         }
       }
     }
 
-    if (authzs.isEmpty || roles.isEmpty || authzs.exists(_.id == "no_rights")) {
+    if (authzs.isEmpty || roles.isEmpty || authzs.contains(AuthorizationType.NoRights)) {
```

Comment:
keep the types! More info, more efficient, avoid useless parse/serialise roundtrip with errors to manage